superset/contrib/docker/Dockerfile

```diff
@@ -24,48 +51,42 @@ ENV LANG=C.UTF-8 \
 
 RUN apt-get update -y
 
-# Install dependencies to fix `curl https support error` and `elaying package configuration warning`
-RUN apt-get install -y apt-transport-https apt-utils
-
+# Install git as it is used for requirements from Github
+RUN apt-get install -y git
 # Install superset dependencies
-# https://superset.incubator.apache.org/installation.html#os-dependencies
-RUN apt-get install -y build-essential libssl-dev \
-    libffi-dev python3-dev libsasl2-dev libldap2-dev libsasl2-2 libsasl2-modules-gssapi-mit libxi-dev
-
-# Install extra useful tool for development
-RUN apt-get install -y vim less postgresql-client redis-tools
+RUN apt-get install -y libsasl2-dev libsasl2-2 libsasl2-modules-gssapi-mit
 
-# Install nodejs for custom build
-# https://superset.incubator.apache.org/installation.html#making-your-own-build
-# https://nodejs.org/en/download/package-manager/
-RUN curl -sL https://deb.nodesource.com/setup_10.x | bash - \
-    && apt-get install -y nodejs
+RUN if [ "$DEV_BUILD" = "true" ]; \
+        then apt-get install -y curl && \
+        curl -sL https://deb.nodesource.com/setup_10.x | bash - \
+        && apt-get install -y nodejs vim less postgresql-client redis-tools; fi
 
 WORKDIR /home/superset
 
+COPY --from=wheels /wheels /wheels
+
 COPY requirements.txt .
 COPY requirements-dev.txt .
 COPY contrib/docker/requirements-extra.txt .
 
 RUN pip install --upgrade setuptools pip \
-    && pip install -r requirements.txt -r requirements-dev.txt -r requirements-extra.txt \
-    && rm -rf /root/.cache/pip
+        && pip install --no-index --find-links=/wheels \
+        -r requirements.txt -r requirements-dev.txt -r requirements-extra.txt \
```

Comment:
`requirements-dev` are needed only for dev :)
(except db drivers which we should move to another file)

Reply:
yup, this doesn't address #272. The reason is that as you said we should also reorg requirements files such as whether to store our requirements in a `requirements-build.txt` or whatever where we store db drivers and pyHive. I wanted to separate this PR which is more general.